gsrs-module-substance-example/src/test/java/example/substance/validation/StandardNameDuplicateValidatorTest.java

```diff
@@ -0,0 +1,212 @@
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.security.test.context.support.WithMockUser;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+@WithMockUser(username = "admin", roles="Admin")
+@Slf4j
+public class StandardNameDuplicateValidatorTest extends AbstractSubstanceJpaEntityTest {
+
+// @Autowired
+// private SubstanceRepository substanceRepository;
+
+@Autowired
+private GsrsCache cache;
+
+@Autowired
+private TestIndexValueMakerFactory testIndexValueMakerFactory;
+
+@BeforeEach
+
+public void runSetup() throws IOException {
+        // log.trace("runSetup");
+        // seems to work without this
+        // AutowireHelper.getInstance().autowireAndProxy(substanceRepository);
+        // SubstanceDefinitionalHashIndexer hashIndexer = new SubstanceDefinitionalHashIndexer();
+        // AutowireHelper.getInstance().autowire(hashIndexer);
+        // testIndexValueMakerFactory.addIndexValueMaker(hashIndexer);
+        // prevent validations from occurring multiple times
+        // File dataFile = new ClassPathResource("testdumps/rep18.gsrs").getFile();
+        // cache.clearCache();
+        // loadGsrsFile(dataFile);
+        // log.trace("loaded rep18 data file");
+}
+
+        @Test
+        public void testFindInRepository() {
+                // This test is not working as expected.
+                Substance s1 = new Substance();
+                Name name1 = new Name();
+                name1.name ="TEST1";
+                name1.stdName ="TEST1 STD";
+                if (name1.languages == null) {
+                        name1.languages = new EmbeddedKeywordList();
+                }
+                name1.languages.add(new Keyword("en"));
+                name1.languages.add(new Keyword("fr"));
+
+                s1.names.add(name1);
+                substanceRepository.saveAndFlush(s1);
+
+                Substance s2 = new Substance();
+                Name name2 = new Name();
+                name2.name ="TEST2";
+                name2.stdName ="TEST1 STD"; // The 1 is on purpose
+                if (name2.languages == null) {
+                        name2.languages = new EmbeddedKeywordList();
+                }
+                name2.languages.add(new Keyword("en"));
+                name2.languages.add(new Keyword("fr"));
+                s2.names.add(name2);
+                substanceRepository.saveAndFlush(s2);
+                cache.clearCache();
+
+                // Searching/Finding on this!!
+                String testStdName = name2.stdName;
+
+                Assertions.assertEquals(substanceRepository.count(), 2);
+
+                // This doesn't work for me in the test context.
+                // findByNames_StdNameIgnoreCase does seem to work as expected when
+                // testing from the frontend.
+                List<SubstanceRepository.SubstanceSummary> ssl = substanceRepository.findByNames_StdNameIgnoreCase(testStdName);
```

Comment:
The method findByNames_StdNameIgnoreCase in the substanceRepository does not seem to work in the test context.   The code for this seems to be generated automagically by springboot. 

The method does seem to work when running the application as a whole from the UI. 

Perhaps some test configuration needs to be added?